notebooks/Notebook2.md

```diff
@@ -18,17 +31,49 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+I spent a good chunk of this week adapting my Python code into Scala, adding
+some richness to it that takes advantage of Scala's DSL-building features. I
+got pretty stuck while trying to implement a "do for all players" function that
+has a more readable syntax than just using a for loop. The problem is that it's
+very difficult to have the iterator variable accessible in the body of the loop,
+without having an onerous amount of syntax for the user to write.
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I need to finish implementing Hearts using my new syntax, and see where the
+continuing sticking points in my code are. My Scala code is not currently more
+*efficient* than my Python code, in terms of lines written, but it's already
+significantly more readable. I'm really happy with where my interrupts are going,
+but I need to work out the prompt and eachPlayer functions. In class, I need to
+ask Prof Wiedermann why my deal function is requiring parentheses, and how I
```

Comment:
I don't know the solution to that. Maybe the parser requires the parenthesis to know its a function? When its a subfunction it can tell by the fact it is part of a larger object, but when its on its own its tougher